### ISSUE TITLE
`prefer-array-at` produces invalid code

### ISSUE THREAD

**Issue by AriPerkkio** [e18e/eslint-plugin#27]
The `prefer-array-at` incorrectly reports following code and produces broken Javascript on the autofix:

```js
const items = ["One", "Two", "Three"];

items[items.length - 1] = items[items.length - 1].slice(0, 1);
```

After autofix:

```js
const items = ["One", "Two", "Three"];

items.at(-1) = items.at(-1).slice(0, 1);   // <-- This won't run
```

Real world use case at https://github.com/AriPerkkio/ast-v8-to-istanbul/blob/68b3edc85583775f15e92897817d974caf25ceb3/src/location.ts#L162

### Reproduction:

- https://stackblitz.com/~/edit/node-axqbpnwv?file=index.js

1. Check `index.js`, run it with `node index.js`
2. Run `pnpm run lint --fix`
3. Run `node index.js` -> Node errors with `Uncaught ReferenceError: Invalid left-hand side in assignment`

(Errors above are from local env. There's a bit different error message in Stackblitz Webcontainers)

**Comment by 43081j** [comment]
good catch!

i'll fix it shortly 👍
